src/test/compile-fail/dep-graph-variance-alias.rs

```diff
@@ -23,7 +23,7 @@ struct Foo<T> {
     f: T
 }
 
-#[rustc_if_this_changed]
+#[rustc_if_this_changed(Krate)]
```

Comment:
Huh, this looks suspicious.

Reply:
This change doesn't seem right to me. I think we do expect to have to recompute item-variances for `Use` if the type alias changes. I would sort of expect that it would be an input into the `AdtDef` or what have you for `Use`... but I guess it might not be working out that way? Did you dig into what's going on here?

Reply:
The `CrateVariances` dep node is `eval_always`, so it only depends on `Krate` and not on any `Hir` nodes. This in turn means that `ItemVariances` won't transitively depend on any `Hir` nodes, just `Krate`. So there won't be a path from `Hir(TypeAlias)` to `ItemVariances`. This is still fine though since `CrateVariances` always gets recomputed. It is also correct without that because `Krate` has an implicit dependency on everything in a crate.

Reply:
OK, makes sense. Thanks!